jina/peapods/networking.py

```diff
@@ -0,0 +1,57 @@
+from argparse import Namespace
+
+from .. import __default_host__
+from ..helper import get_public_ip, get_internal_ip
+
+
+def get_connect_host(
+    bind_host: str, bind_expose_public: bool, connect_args: Namespace
+) -> str:
+    """
+    Compute the host address for ``connect_args``
+
+    :param bind_host: the ip for binding
+    :param bind_expose_public: True, if bind socket should be exposed publicly
+    :param connect_args: configuration for the host ip connection
+    :return: host ip
+    """
+    from sys import platform
+
+    # by default __default_host__ is 0.0.0.0
+
+    # is BIND at local
+    bind_local = bind_host == __default_host__
+
+    # is CONNECT at local
+    conn_local = connect_args.host == __default_host__
+
+    # is CONNECT inside docker?
+    conn_docker = getattr(
+        connect_args, 'uses', None
+    ) is not None and connect_args.uses.startswith('docker://')
+
+    # is BIND & CONNECT all on the same remote?
+    bind_conn_same_remote = (
+        not bind_local and not conn_local and (bind_host == connect_args.host)
+    )
+
+    if platform in ('linux', 'linux2'):
+        local_host = __default_host__
+    else:
```

Comment:
Should we add an if here to catch other OSs?

Reply:
This is exactly the same code as before, just in a different location. I would not touch it, since I am not a Mac expert.